File: Unit4E.ts
/**
 * @author William
 * @version 1.0.0
 * @date 2025-11-27
 * @fileoverview This program calculates base^exponent using a while loop.
 */

// variables
let base: number = 0;
let exponent: number = 0;
let count: number = 0;
let result: number = 1;

// get base from user
base = parseInt(prompt("Enter the base: ") || "0");

// get exponent from user
exponent = parseInt(prompt("Enter the exponent: ") || "0");

// while loop multiplies base exponent times
count = 0;
while (count < exponent) {
  result = result * base;
  count++;
}

// display result
console.log(base + " raised to the power of " + exponent + " is: " + result);

console.log("\nDone.");

File: Unit4E.go
// Author: William
// Version: 1.0.0
// Date: 2025-11-27
// Fileoverview: This program calculates base^exponent using a while-style loop.

package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {

	// variable declarations
	var baseString string
	var exponentString string
	var base int
	var exponent int
	var count int
	var result int = 1

	reader := bufio.NewReader(os.Stdin)

	// get base from user
	fmt.Print("Enter the base: ")
	baseString, _ = reader.ReadString('\n')
	baseString = strings.TrimSpace(baseString)
	base, _ = strconv.Atoi(baseString)

	// get exponent from user
	fmt.Print("Enter the exponent: ")
	exponentString, _ = reader.ReadString('\n')
	exponentString = strings.TrimSpace(exponentString)
	exponent, _ = strconv.Atoi(exponentString)

	// while loop to multiply base exponent times
	count = 0
	for count < exponent {
		result = result * base
		count++
	}

	// print result
	fmt.Printf("%d raised to the power of %d is: %d\n", base, exponent, result)

	fmt.Println("\nDone.")
}
